frame/support/procedural/src/pallet/expand/constants.rs

```diff
@@ -66,13 +69,20 @@ pub fn expand_constants(def: &mut Def) -> proc_macro2::TokenStream {
 				let value = <Pallet<#type_use_gen>>::#ident();
 				#frame_support::codec::Encode::encode(&value)
 			),
+			metadata_name: const_.metadata_name.clone(),
 		}
 	});
 
 	let consts = config_consts.chain(extra_consts).map(|const_| {
 		let const_type = &const_.type_;
 		let ident = &const_.ident;
-		let ident_str = format!("{}", ident);
+		let metadata_name = &const_.metadata_name;
+		let ident_str = if let Some(metadata_ident) = metadata_name {
+			format!("{}", metadata_ident)
+		} else {
+			format!("{}", ident)
+		};
```

Comment:
This may also be simpler, and we can remove the line above
```suggestion
		let ident_str = format!("{}", const_.metadata_name.unwrap_or(const_.ident));
```